### PR TITLE
Add 0.41.0 upgrade routine

### PR DESCRIPTION
**Pull request by VineethReddy02**
Fixes: #623

### CODE REVIEW

File: pkg/collector/upgrade/v0_41_0.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
+	"github.com/open-telemetry/opentelemetry-operator/pkg/collector/adapters"
+)
+
+func upgrade0_41_0(cl client.Client, otelcol *v1alpha1.OpenTelemetryCollector) (*v1alpha1.OpenTelemetryCollector, error) {
+	cfg, err := adapters.ConfigFromString(otelcol.Spec.Config)
+	if err != nil {
+		return otelcol, fmt.Errorf("couldn't upgrade to v0.39.0, failed to parse configuration: %w", err)
```
Comment (jpkrohling) [inline pkg/collector/upgrade/v0_41_0.go:31]:
```suggestion
		return otelcol, fmt.Errorf("couldn't upgrade to v0.41.0, failed to parse configuration: %w", err)
```

File: pkg/collector/upgrade/v0_41_0_test.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    metrics:
+      exporters:
+      - nop
+      receivers:
+      - otlp
+`, res.Spec.Config)
+
+	assert.Equal(t, "upgrade to v0.41.0 has re-structured the cors_allowed_origins inside otlp "+
+		"receiver config according to the upstream otlp receiver changes in 0.41.0 release", res.Status.Messages[0])
+	assert.Equal(t, "upgrade to v0.41.0 has re-structured the cors_allowed_headers inside otlp "+
```
Comment (jpkrohling) [inline pkg/collector/upgrade/v0_41_0_test.go:84]: Why are there two entries here?

**Review by jpkrohling — APPROVED** [review]

File: pkg/collector/upgrade/v0_41_0_test.go (hunk 2 above)
Comment (VineethReddy02) [reply pkg/collector/upgrade/v0_41_0_test.go:84]: Adding entries for both `cors_allowed_headers` and `cors_aloowed_origins` few users might both or either one of the,

Comment (jpkrohling) [reply pkg/collector/upgrade/v0_41_0_test.go:84]: Is it possible then to qualify the messages with the part that got changed?

Comment (VineethReddy02) [reply pkg/collector/upgrade/v0_41_0_test.go:84]: yup the message includes the `cors_allowed_headers` which is added during the upgrade.

Comment (jpkrohling) [reply pkg/collector/upgrade/v0_41_0_test.go:84]: True, I missed that part, sorry!

**Comment by jpkrohling** [comment]
Feel free to merge whenever the CI is green.

**Pull request merged**